library/src/scala/util/control/NonLocalReturns.scala

```diff
@@ -7,8 +7,18 @@ package scala.util.control
  *     import scala.util.control.NonLocalReturns.*
  *
  *     returning { ... throwReturn(x) ... }
+ *
+ *  This API has been deprecated. Its functionality is better served by
+ *
+ *   - `scala.util.boundary` in place of `returning`
+ *   - `scala.util.break` in place of `throwReturn`
+ *
+ *  The new abstractions work with plain `RuntimeExceptions` and are more
+ *  performant, since returns within the scope of the same method can be
+ *  rewritten by the compiler to jumps.
  */
 object NonLocalReturns {
+  @deprecated("Use scala.util.boundary.Break instead", "3.3")
```

Comment:
We should not deprecate this before the new one is stable. 

Do we need a SIP to deprecate this feature?

Reply:
I assume we will also deprecate `scala.util.control.Breaks` as this is a more flexible version of that abstraction.

Reply:
yes, that has to be a PR against scala/scala.